### PR TITLE
test(jQuery): Run tests with jQuery 3

### PR DESCRIPTION
**Pull request by mgol**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
A test update.

**What is the current behavior? (You can also link to an open issue here)**
Currently Angular is tested with jQuery 2.1 & 2.2. jQuery 2.2 is used for e2e tests.

**What is the new behavior (if this is a feature change)?**
Angular will be tested with jQuery 2.1, 2.2 & 3.0; the latest one will be used for e2e tests.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] ~~Tests for the changes have been added (for bug fixes / features)~~
- [x] ~~Docs have been added / updated (for bug fixes / features)~~

**Other information**:

### CODE REVIEW

File: test/ng/directive/selectSpec.js [hunk 1]
```diff
@@ -1888,7 +1888,14 @@ describe('select', function() {
...
             expect(optionElements.length).toEqual(1);
             expect(scope.obj.value).toEqual([]);
-            expect(element.val()).toBe(null);
+
+            // Cover both jQuery 3.x ([]) and 2.x (null) behavior.
+            var val = element.val();
+            if (val === null) {
+              val = [];
+            }
+            expect(val).toEqual([]);
```
Comment (gkalpak) [inline test/ng/directive/selectSpec.js]: Nitpick: I would prefer `expect(element.val()).toBeOneOf(null, [])` for readability.

File: test/ng/directive/selectSpec.js [hunk 2]
```diff
@@ -1888,7 +1888,14 @@ describe('select', function() {
             optionElements = element.find('option');
             expect(optionElements.length).toEqual(1);
             expect(scope.obj.value).toEqual([]);
-            expect(element.val()).toBe(null);
+
+            // Cover both jQuery 3.x ([]) and 2.x (null) behavior.
+            var val = element.val();
```
Comment (Narretz) [inline test/ng/directive/selectSpec.js]: How about `toEqual(null || [])`? Does that work?

Comment (mgol) [inline test/ng/directive/selectSpec.js]: `toEqual(null || [])` is the same as `toEqual([])`. :-) The expression will be computed before passing it to `toEqual`.

**Comment by Narretz** [comment]
Most of the data in the jquery karma conf could be moved into a shared file.
Otherwise LGTM

File: test/ng/directive/selectSpec.js (hunk 2 above)
Comment (gkalpak) [inline test/ng/directive/selectSpec.js]: We have custom `toBeOneOf()` matcher. If we only had `toEqualOneOf()` :stuck_out_tongue:

**Comment by gkalpak** [comment]
Looking at the travis logs, it seems that `tests:jquery-2.2` ran 700 tests less that the other jquery versions. This doesn't seem to be the case on master. Any idea why?

**Comment by mgol** [comment]
@gkalpak Could you be more specific? I don't see such a divergence anywhere.

**Comment by gkalpak** [comment]
In the travis log of this PR's `test:unit` job I see:

jqLite: [42736 tests][1]
jQuery (3.x): [42820 tests][2]
jQuery-2.2: [42120 tests][3]
jQuery-2.1: [42820 tests][4]

(Different number of tests for `jQuery (3.x)`/`jQuery-2.1` vs `jQuery-2.2`.)

--
In the travis log of the latest master build's `test:unit` job I see:

jqLite: [41660 tests][5]
jQuery (2.2): [42884 tests][6]
jQuery-2.1: [42884 tests][7]

(Same number of tests for `jQuery (2.2)` vs `jQuery-2.1`.)

[1]: https://travis-ci.org/angular/angular.js/jobs/142738536#L1110
[2]: https://travis-ci.org/angular/angular.js/jobs/142738536#L1690
[3]: https://travis-ci.org/angular/angular.js/jobs/142738536#L2259
[4]: https://travis-ci.org/angular/angular.js/jobs/142738536#L2837

[5]: https://travis-ci.org/angular/angular.js/jobs/142756002#L1073
[6]: https://travis-ci.org/angular/angular.js/jobs/142756002#L1651
[7]: https://travis-ci.org/angular/angular.js/jobs/142756002#L2230

**Comment by mgol** [comment]
@gkalpak These results are weird. I updated jQuery to 3.1.0 and rebased my PR and now the numbers don't diverge that much; in particular, [jQuery 2.1](https://travis-ci.org/angular/angular.js/jobs/143987356#L7269) has the same number of tests run as [jQuery 2.2](https://travis-ci.org/angular/angular.js/jobs/143987356#L6689)). On the other hand, on one of the commits on master I see [42794 tests run on jQuery 2.2](https://travis-ci.org/angular/angular.js/jobs/143933600#L1666) but [only 39294 on jQuery 2.1](https://travis-ci.org/angular/angular.js/jobs/143933600#L2201).

It seems like some of the tests was being skipped by random (or the total sum is being calculated incorrectly).

In any case, this doesn't seem related to changes in this PR.

**Comment by gkalpak** [comment]
> It seems like some of the tests was being skipped by random (or the total sum is being calculated incorrectly).

I hope it is the latter. It will be very worrying, if it is the former :worried:

**Comment by gkalpak** [comment]
After comparing a couple of master builds, it turns out that the diviations are all related to Safari (for which the total number of tests are always reported as 0 as well). There is obviously some issue with Safari (or some Safari launcher/adapter) wrt to test counts.

In any case, not related to this PR as you said.

LGTM

**Comment by mgol** [comment]
> Most of the data in the jquery karma conf could be moved into a shared file.

I extracted the shared parts.

**Pull request closed without merging**